### DISCUSSION TITLE
When working with Vue it does not throw correct bugs during runtime, just crashes

### DISCUSSION THREAD

**Discussion by maribox** [alex8088/electron-vite#126] [Q&A] [answered]
Hey!
I am using Vue with the composition API and want to send a message from renderer to main.
If I use the following example code, without that functionality it compiles correctly and showes the h1 and button:
```
<script setup lang="ts">
</script>

<template>
  <h1>MAIN</h1>
  <button>Quit</button>
</template>

<style lang="less">
@import './assets/css/styles.less';
</style>

```

But as soon as I try this it does not show anything in the Browser and throws some weird errors, that aren't connected to my program

```
<script setup lang="ts">
import { ipcRenderer } from 'electron'
function greet(event): void {
    if (event) {
        ipcRenderer.invoke('close')
    }
}
</script>

<template>
  <h1>MAIN</h1>
  <button @click="greet">Quit</button>
</template>

<style lang="less">
@import './assets/css/styles.less';
</style>
```

The errors thrown: 
![image](https://user-images.githubusercontent.com/70728036/222981343-dbed96b2-e274-4415-99bc-67f8e4384180.png)

The code in which the error is thrown (not mine, is this from electron-vite?):
```
const fs = require('fs');
const path = require('path');

const pathFile = path.join(__dirname, 'path.txt'); // <-- this line is the one in the error. 

function getElectronPath () {
  let executablePath;
  if (fs.existsSync(pathFile)) {
    executablePath = fs.readFileSync(pathFile, 'utf-8');
  }
  if (process.env.ELECTRON_OVERRIDE_DIST_PATH) {
    return path.join(process.env.ELECTRON_OVERRIDE_DIST_PATH, executablePath || 'electron');
  }
  if (executablePath) {
    return path.join(__dirname, 'dist', executablePath);
  } else {
    throw new Error('Electron failed to install correctly, please delete node_modules/electron and try installing again');
  }
}

module.exports = getElectronPath();
```

It doesn't show me what is wrong or even if something's wrong with my code, just crashes with these errors.

Can somebody help me here? Is this a bug with electron-vite?

**Comment by alex8088** [comment, marked as answer]
not support `nodeIntegration`, should be

`window.electron.ipcRenderer`

learn more about https://evite.netlify.app/guide/dev.html
